### PR TITLE
[SIE-88] convert Webcam snapshots to file

### PR DESCRIPTION
**Pull request by alecherryy**
The BE is still undergoing some updates and changes, so some of the JS will have to change at some point. However, I enabled the React component to send to the server either a Webcam screenshot or a file.

You can view the Jira ticket [here](https://cs6510.atlassian.net/browse/SIE-88).

### CODE REVIEW

File: src/firebase/api/gcp-utils.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const uploadImageToStorage = async (
   const storageRef = storage.ref();
   const rawImageBucketRef = storageRef.child(bucket);
   const newImageRef = rawImageBucketRef.child(imagePath);
-  newImageRef.put(image).then((snapshot) => {
+  return newImageRef.put(image).then((snapshot) => {
```
Comment (alecherryy) [inline src/firebase/api/gcp-utils.js:20]: I believe this file is still undergoing some development work from the BE.

File: src/stories/components/FileUpload/FileUpload.js [hunk 2]
```diff
@@ -13,20 +13,27 @@ import PropTypes from 'prop-types';
  * )
  */
 export const FileUpload = ({ onChange }) => {
+  const setInputValue = (e) => {
+    e.target.previousSibling.innerHTML = e.target.files.item(0).name;
+  };
+
```
Comment (alecherryy) [inline src/stories/components/FileUpload/FileUpload.js:19]: Moved this here to clean up the `UploadPhoto` template.

File: src/stories/components/UploadPhoto/UploadPhoto.js [hunk 3]
```diff
@@ -21,47 +21,66 @@ import ImageService from '../../../firebase/CRUDServices/imageService';
...
     }
+    // call gcp util function
+    uploadImageToStorage(
+      userId, experimentId, file, 'sie-raw-images',
+    ).then((response) => {
+      // REMOVE EVENTUALLY
+      switch (response.status) {
+      case '201':
+        // UPDATE WITH BETTER RESPONSE EVENTUALLY
+        return;
```
Comment (alecherryy) [inline src/stories/components/UploadPhoto/UploadPhoto.js:69]: Once the BE is ready, we'll have to include some additional functionality.

File: src/stories/components/UploadPhoto/UploadPhoto.js [hunk 4]
```diff
@@ -21,47 +21,66 @@ import ImageService from '../../../firebase/CRUDServices/imageService';
...
-    if (document.getElementById('file-upload__input')) {
-      const name = document.getElementById('file-upload__input');
-      const selectedFile = name.files.item(0).name;
-      document.getElementById('file-upload__selected').innerHTML = selectedFile;
+    if (image) {
+      fetch(image).then((response) => response.blob())
+        .then((blob) => {
+          const file = new File([blob], 'photo', { type: 'image/jpeg' });
+          setFile(file);
+        });
```
Comment (alecherryy) [inline src/stories/components/UploadPhoto/UploadPhoto.js:59]: Convert the webcam snapshot to a file.

File: src/firebase/api/gcp-utils.js (hunk 1 above)
Comment (alecherryy) [reply src/firebase/api/gcp-utils.js:20]: Update: there's a PR open to update the `gcp-utils.js` file, once that's merged, I can update the template.

**Review by cmutuc25 — CHANGES_REQUESTED** [review]

File: src/stories/components/UploadPhoto/UploadPhoto.js [hunk 5]
```diff
@@ -76,7 +95,10 @@ export const UploadPhoto = () => {
               modifierClasses="button--small button--secondary"
               isButton={true}
               text="Take a photo"
-              onClick={() => capturePhoto()}
+              onClick={() => {
+                capturePhoto();
+                convertPhotoToFile();
```
[the remark this answers is not in the document]
Comment (alecherryy) [reply src/stories/components/UploadPhoto/UploadPhoto.js:100]: AH good catch.

File: src/stories/components/UploadPhoto/UploadPhoto.js (hunk 4 above)
Comment (alecherryy) [reply src/stories/components/UploadPhoto/UploadPhoto.js:59]: In the state:
```
const [file, setFile] = useState('');
const [image, setImage] = useState('');
```

Comment (cmutuc25) [reply src/stories/components/UploadPhoto/UploadPhoto.js:59]: oh okay, like i was wondering if it was stored in browser memory or local storage but I think state is stored in local storage. Not a big deal I was just curious

**Review by cmutuc25 — APPROVED** [review]

**Pull request merged**